app/Http/Controllers/Api/LinkController.php

```diff
@@ -26,25 +27,32 @@ public function index()
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function show($id)
+    public function show(Request $request, $id)
     {
-        $link = Link::where('id', $id)->first();
-
-        if(count($link) > 0){
-          return $link;
-        }else{
-          return response('Link Not Found', 404);
-        }
+      $user = $request->user();
+      $link = Link::where('id', $id)->firstOrFail();
+      return $link;
     }
 
     /**
-     * Store a newly created link in storage.
+     * Update the specified resource in storage.
      *
      * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function store(Request $request)
+    public function update(EditLink $request, $id)
     {
+      $user = $request->user();
+      $link = Link::where('id', $id)->firstOrFail();
+      $data = $request->all();
+
+      $link->name = $data['name'];
+      $link->url = $data['url'];
+      $link->description = $data['description'];
+
+      $link->save();
 
+      return $link;
```

Comment:
this looks good... but again you get the user but i don't see you doing anything with it